<launch>
	<!-- declaration of arguments -->
	<arg name="robot_number" default="0" doc="Number of the robot we want to launch."/>
	<arg name="debugObstacles" default="0" doc="Verbose Debugging for OPUSim_Obstacles."/>
	<arg name="debugRSO" default="0" doc="Verbose Debugging for OPUSim_RelativeSwarmOdometry."/>
	
	<arg name="emergencyBreak" default="0" doc="Everything will be launched except that the cmd_vel-related topic is publishing 0."/>
	<arg name="tresholdDistAccount" default="0.6" doc="Treshold for the distance of obstacles that we take into account for obstacle avoidance."/>
	
	
	<node pkg="OPUSim" type="OPUSim_OmniView" name="OPUSim_OmniView">
		<rosparam param="robot_number" subst_value="True">$(arg robot_number)</rosparam>
	</node>
	
	<node pkg="OPUSim" type="OPUSim_Obstacles" name="OPUSim_Obstacles">
		<rosparam param="robot_number" subst_value="True">$(arg robot_number)</rosparam> 
		<rosparam param="debug" subst_value="True">$(arg debugObstacles)</rosparam> 
	</node>
	
	<node pkg="OPUSim" type="OPUSim_RelativeSwarmOdometry" name="OPUSim_RelativeSwarmOdometry">
		<rosparam param="robot_number" subst_value="True">$(arg robot_number)</rosparam> 
		<rosparam param="debug" subst_value="True">$(arg debugRSO)</rosparam> 
	</node>
	
	<!--
	<node pkg="OPUSim" type="OPUSim_ControlLaw" name="OPUSim_ControlLaw">
		<rosparam param="robot_number" subst_value="True">$(arg robot_number)</rosparam>
		<rosparam param="emergencyBreak" subst_value="True">$(arg emergencyBreak)</rosparam>
		<rosparam param="tresholdDistAccount" subst_value="True">$(arg tresholdDistAccount)</rosparam>
	</node>
	-->
	
</launch>
